packages/plugin-ext/src/main/browser/view/tree-views-main.tsx

```diff
@@ -0,0 +1,236 @@
+        }
+
+        return {
+            id: item.id,
+            parent: undefined,
+            name: item.label,
+            icon: item.icon,
+            description: item.tooltip,
+            visible: true,
+            selected: false,
+            expanded,
+            children: []
+        };
+    }
+
+    createFileNode(item: TreeViewItem): TreeViewFileNode {
+        return {
+            id: item.id,
+            name: item.label,
+            icon: item.icon,
+            description: item.tooltip,
+            parent: undefined,
+            visible: true,
+            selected: false,
+        };
+    }
+
+    /**
+     * Creates TreeNode
+     *
+     * @param item tree view item from the ext
+     */
+    createTreeNode(item: TreeViewItem): TreeNode {
+        if ('collapsibleState' in item) {
+            if (TreeViewItemCollapsibleState.Expanded === item.collapsibleState) {
+                return this.createFolderNode(item);
+            } else if (TreeViewItemCollapsibleState.Collapsed === item.collapsibleState) {
+                return this.createFolderNode(item);
+            }
+        }
+
+        return this.createFileNode(item);
+    }
+
+    async resolveChildren(itemId: string): Promise<TreeNode[]> {
+        const children = await this.proxy.$getChildren(this.treeViewId, itemId);
+
+        if (children) {
+            return children.map(value => this.createTreeNode(value));
+        }
+
+        return [];
+    }
+
+}
+
+@injectable()
+export class TreeViewWidget extends TreeWidget {
+
+    constructor(
+        @inject(TreeProps) readonly treeProps: TreeProps,
+        @inject(TreeModel) readonly model: TreeModel,
+        @inject(ContextMenuRenderer) readonly contextMenuRenderer: ContextMenuRenderer,
+        @inject(TreeViewDataProviderMain) readonly dataProvider: TreeViewDataProviderMain) {
+
+        super(treeProps, model, contextMenuRenderer);
+
+        setTimeout(() => {
+            this.showDummyNode();
+        }, 1000);
+    }
+
+    public up() {
+        this.updateRows();
+
+        // Need to wait for 20 miliseconds until rows become updated.
+        setTimeout(() => {
+            ReactDOM.render(<React.Fragment>{this.render()}</React.Fragment>, this.node, () => this.onRender.dispose());
+        }, 20);
+    }
+
+    showDummyNode() {
+        const node = {
+            id: 'root-item',
+            parent: undefined,
+            name: 'IDE2 Team',
+            description: 'IDE2 Team Participants',
+            visible: true,
+            expanded: true,
+            selected: false,
+            children: []
+        };
+
+        this.model.root = node;
+    }
+
+    renderIcon(node: TreeNode, props: NodeProps): React.ReactNode {
+        // return super.renderIcon(node, props);
+        if (node.icon) {
+            return <div className={'fa ' + node.icon + ' tree-view-icon'}></div>;
+        }
+
+        // tslint:disable-next-line:no-null-keyword
+        return null;
```

Comment:
Why do you use `null`? `undefined` is not an option?

Reply:
This method have been copied from the parent class which has such stub.
Anyway it's not a problem to change it on undefined.

Reply:
As I see, it should be `undefined`, I was not aware of it either. See https://github.com/theia-ide/theia/issues/3181.